### PR TITLE
Fix sendrecv ut

### PR DESCRIPTION
**Pull request by typhoonzero**
fix https://github.com/PaddlePaddle/Paddle/issues/6978
fix https://github.com/PaddlePaddle/Paddle/issues/6893

### CODE REVIEW

File: paddle/operators/recv_op.cc [hunk 1]
```diff
@@ -57,7 +59,10 @@ class RecvOp : public framework::OperatorBase {
     }
   }
 
-  virtual ~RecvOp() {
+  void Stop() override {
```
Comment (Yancey1989) [inline paddle/operators/recv_op.cc]: Maybe we also need the destructor function, and call `Stop()` in it.

Comment (typhoonzero) [reply paddle/operators/recv_op.cc]: No. `recv_op`'s `Run` may run in another thread (in unit test), `Stop` must be called out side dtor

Comment (Yancey1989) [reply paddle/operators/recv_op.cc]: So we also need to add the interface to `pybind` , maybe we coud do that in another PR.

Comment (typhoonzero) [reply paddle/operators/recv_op.cc]: Yep. We can add it later when running in different clusters.

**Pull request merged**